src/ds5/ds5-device.cpp

```diff
@@ -770,6 +770,19 @@ namespace librealsense
                     RS2_OPTION_ASIC_TEMPERATURE));
         }
 
+        if ((val_in_range(pid, { RS455_PID })) && (_fw_version >= firmware_version("5.12.11.0")))
```

Comment:
Wait... we have thermal monitor only in D455?

Reply:
yes

Reply:
Then isn't the proper way to do this via polymorphism?